salt/modules/napalm_network.py

```diff
@@ -1982,158 +1966,112 @@ def load_template(
+            # of 1 element (for the iteration below)
+            template_hash = [template_hash]
+        elif (
+            template_hash
+            and isinstance(template_hash, six.string_types)
+            and (
+                template_hash.startswith("salt://")
+                or template_hash.startswith("file://")
+            )
+        ):
+            # If the template hash is a file URI, then provide the same value
+            # for each of the templates in the list, as probably they all
+            # share the same hash file, otherwise the user should provide
+            # this as a list
+            template_hash = [template_hash] * len(template_name)
+        elif not template_hash:
+            template_hash = [None] * len(template_name)
+        for tpl_index, tpl_name in enumerate(template_name):
+            tpl_hash = template_hash[tpl_index]
+            tpl_hash_name = template_hash_name[tpl_index]
+            _rand_filename = __salt__["random.hash"](tpl_name, "md5")
+            _temp_file = __salt__["file.join"]("/tmp", _rand_filename)
+            _managed = __salt__["file.get_managed"](
+                name=_temp_file,
+                source=tpl_name,
+                source_hash=tpl_hash,
+                source_hash_name=tpl_hash_name,
+                user=None,
+                group=None,
+                mode=None,
+                attrs=None,
                 template=template_engine,
                 context=context,
                 defaults=defaults,
                 saltenv=saltenv,
+                skip_verify=skip_verify,
             )
-            if not isinstance(_rendered, six.string_types):
-                if "result" in _rendered:
-                    _loaded["result"] = _rendered["result"]
-                else:
-                    _loaded["result"] = False
-                if "comment" in _rendered:
-                    _loaded["comment"] = _rendered["comment"]
-                else:
-                    _loaded["comment"] = "Error while rendering the template."
-                return _loaded
-        else:
-            # render the file - either local, either remote
-            if not isinstance(template_name, (list, tuple)):
-                template_name = [template_name]
-            if template_hash_name and not isinstance(template_hash_name, (list, tuple)):
-                template_hash_name = [template_hash_name]
-            elif not template_hash_name:
-                template_hash_name = [None] * len(template_name)
-            if (
-                template_hash
-                and isinstance(template_hash, six.string_types)
-                and not (
-                    template_hash.startswith("salt://")
-                    or template_hash.startswith("file://")
-                )
+            if not isinstance(_managed, (list, tuple)) and isinstance(
+                _managed, six.string_types
```

Comment:
We could take the time and stop using `six.string_types` in favor or `str` on places where code is updated.